daemon/commit.go

```diff
@@ -126,6 +127,10 @@ func (daemon *Daemon) Commit(name string, c *backend.ContainerCommitConfig) (str
 		return "", err
 	}
 
```

Comment:
And I think we should check if the container being commit is `RemovalInProgress` or `Dead`, A container which is  `RemoveInProgress` or `Dead` may have removed the rw layer, commit it will cause unexpected result.